src/redux/users/users.saga.js

```diff
@@ -0,0 +1,97 @@
+import { takeEvery, call, put } from 'redux-saga/effects';
+
+import { config } from '../../configs';
+
+import {
+  getAllUsers,
+  getUserById,
+  deleteUser,
+  switchUserStatus
+} from './users.operations';
+
+import {
+  setUsers,
+  setUser,
+  setUserError,
+  setUsersLoading
+} from './users.actions';
+
+import {
+  GET_USERS,
+  GET_USER,
+  DELETE_USER,
+  UPDATE_USER_STATUS
+} from './users.types';
+
+import {
+  setSnackBarSeverity,
+  setSnackBarStatus,
+  setSnackBarMessage
+} from '../snackbar/snackbar.actions';
+
+const { SUCCESS_DELETE_STATUS, SUCCESS_UPDATE_STATUS } = config.statuses;
+
+function* handleUsersLoad() {
+  try {
+    yield put(setUsersLoading(true));
+    const users = yield call(getAllUsers, null);
+    yield put(setUsers(users));
+    yield put(setUsersLoading(false));
+  } catch (err) {
+    yield call(handleUsersError, err);
+  }
+}
+
+function* handleUserLoad({ payload }) {
+  try {
+    yield put(setUsersLoading(true));
+    const user = yield call(getUserById, payload);
+    yield put(setUser(user));
+    yield put(setUsersLoading(false));
+  } catch (err) {
+    yield call(handleUsersError, err);
+  }
+}
+
+function* handleUsersDelete({ payload }) {
+  try {
+    yield put(setUsersLoading(true));
+    yield call(deleteUser, payload);
+    yield call(handleUsersLoad, null);
+    yield call(handleSnackBarSuccess, SUCCESS_DELETE_STATUS);
```

Comment:
I think we also need to redirect the client to users list after deleting because the user is deleted and we should not show its data anymore

Reply:
l used delete function at the user route so l just update users list